Shopipy.GiftcardManagement/Controllers/GiftCardController.cs

```diff
@@ -0,0 +1,83 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Shopipy.Persistence.Models;
+using Shopipy.GiftCardManagement.DTOs;
+using Shopipy.Shared;
+using Shopipy.Shared.DTOs;
+using Shopipy.Shared.Services;
+
+namespace Shopipy.GiftCardManagement.Controllers;
+
+[Authorize]
+[Route("businesses/{businessId}/giftcards")]
+[ApiController]
+public class GiftCardController(IGiftCardService _giftCardService, IMapper _mapper) : ControllerBase
+{
+    [HttpPost]
+    public async Task<IActionResult> CreateGiftCard(GiftCardRequestDTO dto, int businessId)
+    {
+        var giftCard = _mapper.Map<GiftCard>(dto);
+
+        var createdGiftCard = await _giftCardService.CreateGiftCardAsync(giftCard, businessId);
+
+        var giftCardResponseDTO = _mapper.Map<GiftCardResponseDTO>(createdGiftCard);
+
+        return CreatedAtAction(nameof(GetGiftCardById), new { businessId, giftCardId = createdGiftCard.GiftCardId }, giftCardResponseDTO);
+    }
+
+    [HttpGet]
+    public async Task<IActionResult> GetAllGiftCards(int businessId, int? top = null, int? skip = null)
+    {
+        var giftCards = await _giftCardService.GetAllGiftCardsOfBusinessAsync(businessId, top, skip);
+        var count = await _giftCardService.GetGiftCardCountAsync(businessId);
+
+        return Ok(new PaginationResultDto<GiftCardResponseDTO>
+        {
+            Data = giftCards.Select(_mapper.Map<GiftCardResponseDTO>),
+            Count = count
+        });
+    }
+
```

Comment:
Add batch creation for gift cards (e.g. use case: a business owner wants to create 100 gift cards to distribute as a promotion)

Reply:
Don’t create new requirements

Reply:
> Don’t create new requirements

agreed